draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -2157,9 +2160,44 @@ the subsection that gives the corresponding parameters.
 | SECP256k1                 | {{suites-secp256k1}} |
 | BLS12-381                 | {{suites-bls12381}}  |
 
+## Suite ID naming conventions {#suiteIDformat}
+
+New Suite IDs MUST be constructed as follows:
+
+    CURVE_ID || "-" || HASH_ID || "-" || MAP_ID || "-" || ENC_TYPE || "-"
+
+The fields CURVE\_ID, HASH\_ID, MAP\_ID, and ENC\_TYPE are
+ASCII-encoded strings of at most 64 characters each.
+Fields can contain only ASCII characters between 0x21 and 0x7E (inclusive)
+other than hyphen and underscore (i.e., 0x2d, and 0x5f).
+As indicated above, each field (including the last) is followed by a hyphen
+("-", ASCII 0x2d); this makes it easier to create prefix-free Suite IDs.
+
+Fields MUST be chosen as follows:
+
+- CURVE\_ID: a human-readable representation of the target elliptic curve.
+
+- HASH\_ID: a human-readable representation of the hash function used in
+  hash\_to\_base ({{hashtobase}}).
+
+- MAP\_ID: a human-readable representation of the map\_to\_curve function
+  ({{mappings}}).
+
+- ENC\_TYPE: a string indicating whether the suite represents
+  a hash\_to\_curve or an encode\_to\_curve operation ({{roadmap}}).
+
+    - If ENC\_TYPE begins with "RO", the suite uses hash\_to\_curve.
+    - If ENC\_TYPE begins with "NU", the suite uses encode\_to\_curve.
+    - ENC\_TYPE MUST NOT begin with any other string.
+
+The ENC\_TYPE field can be used to encode a suite version number.
+The RECOMMENDED way to do so is to append a string of the form ":Vnn".
+For example, "RO:V02" is an appropriate choice for the second version
```

Comment:
Love the idea of adding the version to an existing field. But why the enc type field? To me this implies changing the entire definition of the random oracle construction, versus making, E.g., curve specific changes?

Reply:
Ah, this is a great question!

In my head, the reason was just that there's nothing else to go in the ENC field other than the two characters "RO" or "NU", so there's tons of room for subfields while still respecting the (somewhat arbitrary) 64 char per field size limit.

(Regarding the size limit: I thought having an explicit size limit would make impls somewhat easier to reason about.)

What if we renamed it to "ENC_MISC" to make this more explicit?

An alternative is to have a separate "misc" field, but then it doesn't seem to make sense to allow 64 chars for the "ENC_TYPE" field, at which point the grammar gets more complex because some fields have different sizes than other fields.

Reply:
>  and it's honestly not obvious that we'll ever use the misc field...

Yeah, the following is very much TIOLI, since it does seem low importance.

A few potential counter points: implementations might assume exactly 2 bytes for `RO` or `NU`. Although following this logic very much leads us into needing a GREASE-like system. 

I mostly just semantically find the idea of versioning the "encoding type" to feel weird. 

> What if we renamed it to "ENC_MISC" and modified the description to make this more explicit?

How about "ENC_VAR" ~ encoding variant?

Reply:
ENC_VAR sounds great to me. @chris-wood?

Reply:
Yep, sounds good to me, too! (Though I would be fine with either. :))

Reply:
OK, changed. Thoughts?